src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -325,7 +320,15 @@ public static Type GetUnderlyingType(Type enumType)
         }
 
         public static TEnum[] GetValues<TEnum>() where TEnum : struct, Enum
-            => (TEnum[])GetValues(typeof(TEnum));
+        {
+            RuntimeType enumType = (RuntimeType)typeof(TEnum);
+            ulong[] ulValues = Enum.InternalGetValues(enumType);
+
+            TEnum[] ret = new TEnum[ulValues.Length];
+            for (int i = 0; i < ulValues.Length; i++)
+                ret[i] = (TEnum)ToObject(enumType, ulValues[i]);
```

Comment:
We have the ToUInt64 generic helper that takes a ulong and produces a TEnum.  Would it be worth adding the inverse and using it here to avoid roundtripping through a boxed object here?  Or you don't think it's worthwhile since there's already the array allocation?

Reply:
The reason for this change is AOT friendliness. The non-generic GetValues method is not AOT friendly since it creates a new array type dynamically.

Yeah, I agree that this can be done in a better way. I am going to undo this change for this PR, and deal with it in next one.

Reply:
Just linking for reference:
https://github.com/dotnet/runtime/pull/44355#discussion_r518947353